Test/Tests/Integration/NodeManager.Tests.ps1

```diff
@@ -0,0 +1,250 @@
+Param (
+    [Parameter(Mandatory=$false)] [string] $TestenvConfFile,
+    [Parameter(Mandatory=$false)] [string] $LogDir = "pesterLogs",
+    [Parameter(ValueFromRemainingArguments=$true)] $UnusedParams
+)
+
+. $PSScriptRoot\..\..\..\CIScripts\Common\Invoke-UntilSucceeds.ps1
+. $PSScriptRoot\..\..\..\CIScripts\Common\Invoke-NativeCommand.ps1
+. $PSScriptRoot\..\..\..\CIScripts\Common\Init.ps1
+. $PSScriptRoot\..\..\..\CIScripts\Common\Aliases.ps1
+. $PSScriptRoot\..\..\..\CIScripts\Testenv\Testenv.ps1
+. $PSScriptRoot\..\..\..\CIScripts\Testenv\Testbed.ps1
+
+. $PSScriptRoot\..\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\..\Utils\ComponentsInstallation.ps1
+. $PSScriptRoot\..\..\Utils\ContrailNetworkManager.ps1
+. $PSScriptRoot\..\..\TestConfigurationUtils.ps1
+. $PSScriptRoot\..\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\..\PesterHelpers\PesterHelpers.ps1
+. $PSScriptRoot\..\..\PesterLogger\PesterLogger.ps1
+. $PSScriptRoot\..\..\PesterLogger\RemoteLogCollector.ps1
+
+$ConfigPath = "C:\ProgramData\Contrail\etc\contrail\contrail-vrouter-nodemgr.conf"
+$LogPath = "C:\ProgramData\contrail\var\log\contrail\contrail-vrouter-nodemgr.log"
+
+function New-NodeMgrConfig {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    $HostIP = Get-PSIPAddress -Session $Session
+
+    $Config = @"
+[DEFAULTS]
+log_local = 1
+log_level = SYS_DEBUG
+log_file = $LogPath
+hostip=$HostIP
+db_port=9042
+db_jmx_port=7200
+
+[COLLECTOR]
+server_list=$($ControllerConfig.Address):8086
+
+[SANDESH]
+introspect_ssl_enable=False
+sandesh_ssl_enable=False
+"@
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        Set-Content -Path $Using:ConfigPath -Value $Using:Config
+    }
+}
+
+function Clear-NodeMgrLogs {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        if(Test-Path -Path $Using:LogPath) {
+            Remove-Item $Using:LogPath -Force
+        }
+    }
+}
+
+function Test-NodeMgrLogs {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        return Test-Path -Path $Using:LogPath
+    }
+
+    return $Res
+}
+
+function Test-NodeMgrConnectionWithController {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    $TestbedHostname = Invoke-Command -Session $Session -ScriptBlock {
+        hostname
+    }
+    $Out = Invoke-RestMethod ("http://" + $ControllerConfig.Address + ":8089/Snh_ShowCollectorServerReq?")
+    $OurNode = $Out.ShowCollectorServerResp.generators.list.GeneratorSummaryInfo.source | Where-Object "#text" -Like "$TestbedHostname"
+
+    if($OurNode) {
+        return $true
```

Comment:
Is casting to boolean possible in this _very good_ language?

Reply:
It is!